### PR TITLE
openssl: provide stock config for ca & req apps

### PR DESCRIPTION
**Pull request by xnox**
Ship openssl.cnf in openssl package, replace old openssl-config
package (just in case), include a separate (hardened) ca.cnf.

Set replaces priority, just in case.

### CODE REVIEW

**Comment by xnox** [comment]
(cc @max-allan @max-allan-cgr will love this)

**Review by dannf — COMMENTED** [review]
Just commenting now, still need to review the fips side to make sure I understand everything before I can +1 in good faith

File: openssl/ca.cnf
Comment (xnox) [reply openssl/ca.cnf:49]: Yes, it did made me double-take this when I read it line-by-line too.

I think fax maybe off, but I will make sure to try to do a pager message to them.

Also it's over 20 years now since netscape died.

File: openssl.yaml
Comment (xnox) [reply openssl.yaml]: strong backwards compat guarantees from openssl; it is part of their stable abi commitment and the same ca.cnf can work from like 1.0.0 days all the way to today.

However, we should review this on major openssl upgrades; as this is ultimately parsed by libcrypto.so / the openssl.yaml build; and thus any new features or breaking changes have to be together with openssl upgrades. Thus in practice a twice yearly review of this; upon openssl major upgrades is warranted here.

The package tests & images tests will also catch if this integration stops working.

File: openssl/ca.cnf
Comment (xnox) [reply openssl/ca.cnf:119]: All good invocations are typically these are overriden.

The stock values of Australia, Some-State, Internet Widgits Pty LTD are very disorienting, and frankly look SUS to an untrained eye.

These values are inline with package/vendor branding elsewhere in openssl/gcc/glibc/openjdk and so on; and will instantly help people identify that these certs were generated in a chainguard-image and will be able to find our support. So it is multi-purpose breadcrumb which also should reduce knee-jerk reactions and anxiety as to who Internet Widgits Pty Ltd are.

File: openssl.yaml
Comment (xnox) [reply openssl.yaml:12]: yes; once this lands openssl-config.apk must no longer be used; which itself is EOL and no longer updated package....... at some point there were no references to openssl-config, and yes I am adding this because they could have crept back in.

Comment (dannf) [reply openssl.yaml:12]: To track: https://github.com/chainguard-dev/internal-dev/issues/7799

**Review by dannf — APPROVED** [review]

**Pull request merged**
